java/ql/lib/semmle/code/java/security/BrokenCryptoAlgorithmQuery.qll

```diff
@@ -15,6 +15,7 @@ private class ShortStringLiteral extends StringLiteral {
 class BrokenAlgoLiteral extends ShortStringLiteral {
   BrokenAlgoLiteral() {
     this.getValue().regexpMatch(getInsecureAlgorithmRegex()) and
+    not this.getValue().regexpMatch(getASecureAlgorithmName()) and
```

Comment:
This ought to be fixed as follows to make sense with `regexpMatch` (`getASecureAlgorithmName` contains, perhaps a bit surprisingly, regexes):

```suggestion
    not this.getValue().regexpMatch(getSecureAlgorithmRegex()) and
```

Nonetheless see my comment above for the full picture.